pkg/operator/minio/controller.go

```diff
@@ -0,0 +1,304 @@
+	appsClient := c.context.Clientset.AppsV1beta2()
+
+	envVars := map[string]string{
+		"MINIO_ACCESS_KEY": spec.Credentials.AccessKey,
+		"MINIO_SECRET_KEY": spec.Credentials.SecretKey}
+
+	podSpec := c.makeMinioPodSpec(name, namespace, minioCtrName, minioCtrImage, spec.Port, envVars, int32(spec.Storage.NodeCount), spec.ObjectStoreDataDir)
+
+	nodeCount := int32(spec.Storage.NodeCount)
+	ss := v1beta2.StatefulSet{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels:    map[string]string{"app": minioLabel},
+		},
+		Spec: v1beta2.StatefulSetSpec{
+			Replicas: &nodeCount,
+			Selector: &meta_v1.LabelSelector{
+				MatchLabels: map[string]string{"app": minioLabel},
+			},
+			Template: podSpec,
+			VolumeClaimTemplates: []v1.PersistentVolumeClaim{
+				{
+					ObjectMeta: meta_v1.ObjectMeta{
+						Name:      minioVolumeName,
+						Namespace: namespace,
+					},
+					Spec: v1.PersistentVolumeClaimSpec{
+						AccessModes: []v1.PersistentVolumeAccessMode{v1.ReadWriteOnce},
+						Resources: v1.ResourceRequirements{
+							Requests: v1.ResourceList{
+								v1.ResourceStorage: resource.MustParse(minioStorageGBString),
+							},
+						},
+					},
+				},
+			},
+			ServiceName: name,
+			// TODO: liveness probe
+		},
+	}
+
+	return appsClient.StatefulSets(namespace).Create(&ss)
+}
+
+func (c *MinioController) makeMinioService(name, namespace string, spec miniov1alpha1.ObjectStoreSpec) (*v1.Service, error) {
+	coreV1Client := c.context.Clientset.CoreV1()
+
+	svc, err := coreV1Client.Services(namespace).Create(&v1.Service{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels:    map[string]string{"app": minioLabel},
+		},
+		Spec: v1.ServiceSpec{
+			Type:     v1.ServiceTypeNodePort,
+			Selector: map[string]string{"app": minioLabel},
+			Ports:    []v1.ServicePort{{Port: spec.Port}},
+		},
+	})
+
+	return svc, err
+}
+
+func (c *MinioController) onAdd(obj interface{}) {
+	objectstore := obj.(*miniov1alpha1.ObjectStore).DeepCopy()
+
+	// Create the headless service.
+	logger.Infof("Creating Minio headless service %s.", objectstore.Name)
+	_, err := c.makeMinioHeadlessService(objectstore.Name, objectstore.Namespace, objectstore.Spec)
+	if err != nil {
+		logger.Errorf("failed to create minio headless service: %v", err)
+		return
+	}
+	logger.Infof("Finished creating Minio headless service %s.", objectstore.Name)
+
+	// Create the stateful set.
+	logger.Infof("Creating Minio stateful set %s.", objectstore.Name)
+	_, err = c.makeMinioStatefulSet(objectstore.Name, objectstore.Namespace, objectstore.Spec)
+	if err != nil {
+		logger.Errorf("failed to create minio stateful set: %v", err)
+		return
+	}
+	logger.Infof("Finished creating Minio stateful set %s.", objectstore.Name)
+
+	// Create the nodeport service.
+	svcName := objectstore.Name + "-service"
```

Comment:
could you make the service name just the name of the object store? Since it's a service type, a `service` suffix wouldn't be needed.

Reply:
We won't need to worry about this if we just let the admins expose the service however they want.

Reply:
exposing it internal to the cluster by default still makes sense by default IMO